diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
+        are still two sets of perpendicular grid lines parallel to the {100}
+        directions on each cube face, but the spacing of the grid lines is
+        chosen so that the angles between the grid points on the line 
+        connecting the face centers (<001>) to the edges (<011>) are equal.
+        The maximum angle is also between the <001> directions and the first
+        grid point towards the <011> edges.
+
+        The spherified_corner case is similar to the spherified_edge case, but
+        the spacing of the grid lines is chosen so that the angles between
+        the grid points on the line connecting the face centers to the cube
+        corners (<111>) is equal. The maximum angle in this grid is from the
+        corners to the first grid point towards the cube face centers.
+
+    Returns
+    -------
+    points_in_cartesian : np.array (N,3)
+        Rows are x, y, z where z is the 001 pole direction
+
+    References
+    ----------
+    [1] https://medium.com/game-dev-daily/four-ways-to-create-a-mesh-for-a-sphere-d7956b825db4
+    """
+    # the angle between 001 and 011
+    max_angle = np.arccos(1/np.sqrt(2))
+    # the distance on the cube face from 001 to 011 grid point
+    max_dist = 1
+    if grid_type == "normalized":
+        grid_len = np.tan(np.deg2rad(resolution))
+        steps = math.ceil(max_dist/grid_len)
+        i = np.arange(-steps, steps)/steps
+    elif grid_type == "spherified_edge":
+        steps = math.ceil(np.rad2deg(max_angle)/resolution)
+        k = np.arange(-steps, steps)
+        theta = np.arctan(max_dist)/steps
+        i = np.tan(k*theta)
+    elif grid_type == "spherified_corner":
+        # the angle from 001 to 111
+        max_angle_111 = np.arccos(1/np.sqrt(3))
+        res_111 = np.deg2rad(resolution)
+        steps = math.ceil(max_angle_111/res_111)
+        k = np.arange(-steps, steps)
+        theta = np.arctan(np.sqrt(2))/steps
+        i = np.tan(k*theta)/np.sqrt(2)
+    else:
+        raise ValueError(f"grid type {grid_type} not a valid grid type. "
+                         f"Valid options: normalized, spherified_edge, "
+                         f"spherified_corner.")
+    x, y = np.meshgrid(i, i)
+    x, y = x.ravel(), y.ravel()
+    z = np.ones(x.shape[0])
+    # the grid on all faces of the cube, avoiding overlap of points on edges
+    bottom = np.vstack([-x, -y, -z]).T
+    top = np.vstack([x, y, z]).T
+    east = np.vstack([z, x, -y]).T
+    west = np.vstack([-z, -x, y]).T
+    south = np.vstack([x, -z, y]).T
+    north = np.vstack([-x, z, -y]).T
+    # two corners are missing with this procedure
+    m_c = np.array([[-1, 1, 1],
+                    [1, -1, -1]])
+    # combine
+    all_vecs = np.vstack([bottom, top, east, west, south, north, m_c])
+    return normalize_vectors(all_vecs)
+
+
+def _compose_from_faces(corners, faces, n):
+    """
+    Helper function to refine a grid starting from a platonic solid,
+    adapted from meshzoo [1]
```

Comment:
this reference is absent, I suggest removing it here, adding a See Also to the bottom with the main function (which also has the reference)